### ISSUE TITLE
Patrick Progress 01/08

### ISSUE THREAD

**Issue by arennax** [arennax/emse_redo#1]
# For EMSE paper revision:
 
Two major issues: A. Data Cleaning  B. Fair Comparison

Entire experiments for the above issues have been re-done. New results show as follow:

https://github.com/arennax/emse_redo/blob/master/results/New_Experiment_Results.pdf

Among those results:

## One methods:
16/18:   CART_FLASH
14/18:   CART_DE
07/18:   LP4EE

## Two methods:
17/18:   CART_FLASH + LP4EE
17/18:   CART_DE + CART_FLASH
17/18:   ATLM + CART_FLASH

## Three methods:
18/18:   ATLM + CART_DE + CART + FLASH
18/18:   CART_DE + CART_FLASH + LP4EE

All data have been cleaned followed by **reviewer 2** (F. Sarro)'s comments.
Optimizers have been re-setting to make fair comparison, i.e. same number of evaluations for DE and FLASH, also compare with same number of random choice, followed by **reviewer 3**'s comments. 
(DE: pop=20, iteration=10, total #eval = 20 * 10 = 200)
(FLASH: initial pop = 10, additional evaluation = 190, total #eval = 10 + 190 = 200)
(Random Choice: Pick the best from 200 random selections)

I will finish the point-to-point text task of reviewers' comments in next few days

# For new work:

## (1) Deep Learning Paper:
Automating Intention Mining, TSE’18, Huang

**Related Dataset has been found here:**
https://github.com/tkdsheep/Intention-Mining-TSE/tree/master/data

**Some problem:**
- Running much slower (30min vs 1s)
- Automatic Hyperparameters Tuning? (A greedy way, and results even better than grid search?)
- Compare with default SVM, RF, NB, KNN (No tuning)
- No mention about their DE-SVM (Tuning let 1s + 11min?)
- Bias Metrics (accuracy vs recall?)

**What I can do:**
- Ask for the dataset they used (Important) (**Done**)
- Study the taxonomy of linguistic patterns, Di Sorbo methods
- Study their model: Convolutional NN, Batch Normalization, hyperparameter choice
- Some text mining techniques: VSM, bag-of-words, Word embedding, skip-gram model
- Classification problem metrics
- Reproduce their results (Maybe not)
- Re-try DE-SVM
- Use tuned SVM, RF, NB, KNN (with flash or DE ...)
- Re-compare the performance (different metrics) and cost

## (2) Github Effort Estimation Paper:
Software effort estimation based on open source projects: Case study of Github, IST’17, Qi

**Related Dataset has been found here:**
https://sites.google.com/site/whuxyjfmq/home/see_osp_dataset

**Some problem:**
- Three aspects for project selection (time, team size, business related)?
- Cost metrics (features) used? LOC? Refer from COCOMO-2
- Transform methods for discrete values (sigmoid function)
- Person_month calculating formula?
- 147 samples, ABCART expansion?
- Only java open source projects on Github
- Keywords for effort estimation dataset?

**What I can do:**
- Ask their Data (**Done**)
- Try using their method to collect data and see the experiment results
- Design different aspects for project selection
- Design different required cost metrics (features)
- Different way to test results (not just default methods like CART)
- Test the methods by using other open source projects on Github (Not just Java)

# Some interesting reading stuff

## Effort Estimation
A novel automated approach for software effort estimation based on data augmentation
https://www.researchgate.net/publication/328591967_A_novel_automated_approach_for_software_effort_estimation_based_on_data_augmentation

## Hyperparameter Optimization
Hyperparameter Optimization with SigOpt on AWS
https://aws.amazon.com/solutionspace/sigopt_on_aws1/

## Potential place to apply FLASH and active learning
Feature-interaction aware configuration prioritization
https://dl.acm.org/citation.cfm?id=3275437&dl=ACM&coll=DL

## Deep Learning debating (Stable over unstable?)

**Comment by timm** [comment]
> https://sites.google.com/site/whuxyjfmq/home/see_osp_dataset

oh dear lord I have died and gone to heaven

what are these attributes?

- @attribute LOC numeric
- @attribute EI numeric
- @attribute EO numeric
- @attribute AFP numeric
- @attribute APEX numeric
- @attribute LTEX numeric
- @attribute FILES numeric
- @attribute EFFORT numeric

> Different way to test results (not just default methods like CART)

no need. all you need to do is get their method going and then apply your flash method. low hanging fruit. done and done. easy paper.

**Comment by timm** [comment]
"Sequential Model Optimization for  Effort Estimation of Contemporary Software Projects
"

**Comment by timm** [comment]
now, good todo list. and good seeing so much to done being to done

**Comment by arennax** [comment]
> now, good todo list. and good seeing so much to done being to done
